Upload external contribution data to s3

PaliC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #95747

Context: We want to create a metric panel to track external contributions to the PyTorch repo

This PR creates a daily job to track how many external contributions occurred the day before and uploads it to a s3 collection which is accessible by rockset.

`upload_external_contrib_stats.py` is a python script which grabs the neccesary stats from github and sticks them into an s3 bucket. It is used here to do daily uploads, but can generally be used for larger queries as well.

PaliC committed:
Upload external contribution data to rockset

[ghstack-poisoned]

PaliC committed:
Update on "Upload external contribution data to rockset"




[ghstack-poisoned]

PaliC committed:
Update on "Upload external contribution data to rockset"




[ghstack-poisoned]

kit1980 commented on tools/stats/upload_external_contrib_stats.py:
```diff
@@ -0,0 +1,114 @@
+from upload_stats_lib import upload_to_s3
+import argparse
+
+FILTER_OUT_USERS = set(["raghuramank100", "facebook-github-bot"])
```
I'd update to 
`FILTER_OUT_USERS = set(["pytorchmergebot", "facebook-github-bot", "pytorch-bot[bot]"])`

(raghuramank100 is no longer relevant, also remove `-author:pytorchmergebot` from line 69 and use the one from `FILTER_OUT_USERS` for maintainabilty)

huydhn commented on .github/workflows/upload-contrib-stats.yml:
```diff
@@ -0,0 +1,36 @@
+          echo "Uploading external contribution stats for $(date -v-1d +%F)"
+          python3 -m tools.stats.upload_external_contrib_stats --startDate $(date -v-1d +%F)  
+
+  check-api-rate:
```
You don't need this here I think. This was mainly used to debug rate limit we have with GitHub a while back (then GitHub increased our limit, and the issue seems to go away for now)

PaliC committed:
Update on "Upload external contribution data to s3"


Context: We want to create a metric panel to track external contributions to the PyTorch repo

This PR creates a daily job to track how many external contributions occurred the day before and uploads it to a s3 collection which is accessible by rockset.

`upload_external_contrib_stats.py` is a python script which grabs the neccesary stats from github and sticks them into an s3 bucket. It is used here to do daily uploads, but can generally be used for larger queries as well.


[ghstack-poisoned]

PaliC committed:
Update on "Upload external contribution data to s3"


Context: We want to create a metric panel to track external contributions to the PyTorch repo

This PR creates a daily job to track how many external contributions occurred the day before and uploads it to a s3 collection which is accessible by rockset.

`upload_external_contrib_stats.py` is a python script which grabs the neccesary stats from github and sticks them into an s3 bucket. It is used here to do daily uploads, but can generally be used for larger queries as well.


[ghstack-poisoned]

PaliC committed:
Update on "Upload external contribution data to s3"


Context: We want to create a metric panel to track external contributions to the PyTorch repo

This PR creates a daily job to track how many external contributions occurred the day before and uploads it to a s3 collection which is accessible by rockset.

`upload_external_contrib_stats.py` is a python script which grabs the neccesary stats from github and sticks them into an s3 bucket. It is used here to do daily uploads, but can generally be used for larger queries as well.


[ghstack-poisoned]

PaliC committed:
Update on "Upload external contribution data to s3"


Context: We want to create a metric panel to track external contributions to the PyTorch repo

This PR creates a daily job to track how many external contributions occurred the day before and uploads it to a s3 collection which is accessible by rockset.

`upload_external_contrib_stats.py` is a python script which grabs the neccesary stats from github and sticks them into an s3 bucket. It is used here to do daily uploads, but can generally be used for larger queries as well.


[ghstack-poisoned]

huydhn commented on .github/workflows/upload-contrib-stats.yml:
```diff
@@ -0,0 +1,27 @@
+name: Upload test stats
```
Nit: Let's use another name here, as we sometime use this name to query workflow runs from Rockset.  This name is currently used by the `upload-test-stats` workflow

PaliC committed:
Update upload-contrib-stats.yml

huydhn commented on tools/stats/upload_external_contrib_stats.py:
```diff
@@ -0,0 +1,143 @@
+
+def _fetch_url(
+    url: str,
+    *,
```
Hmm, do you mean to have '*' here?
```suggestion
```

huydhn commented on tools/stats/upload_external_contrib_stats.py:
```diff
@@ -0,0 +1,143 @@
+def _fetch_url(
+    url: str,
+    *,
+    headers: Optional[Dict[str, str]] = None,
```
Would this work?  Then line 23-24 can be removed I think

```suggestion
    headers: Optional[Dict[str, str]] = {},
```

huydhn commented on tools/stats/upload_external_contrib_stats.py:
```diff
@@ -0,0 +1,143 @@
+
+    pr_count = 0
+    users = set([Dict[str, Any]])
+    while period_begin_date < end_date:
```
Just a note: if querying GitHub API manually gets in the way, you could consider using https://github.com/khornberg/octokit.py to streamline the API call

huydhn reviewed (approved):
There are minor comments here and there but overall LGTM!  Let's also wait if Sergii has any additional comment on the script as he's most familiar with the process

huydhn commented on tools/stats/upload_external_contrib_stats.py:
```diff
@@ -0,0 +1,143 @@
+    period_begin_date = start_date
+
+    pr_count = 0
+    users = set([Dict[str, Any]])
```
Should it be just a set of string? as you only add the username here `users.add(u)`.  If you have done it yet, let's `lintrunner tools/stats/upload_external_contrib_stats.py` to fix any linting issue around

PaliC replied:
given there aren't any errors, I assume users are given as such a dict.

kit1980 replied:
Probably no errors because this file is not type-checked?
From the code it looks like set of stings indeed.

PaliC committed:
Update on "Upload external contribution data to s3"


Context: We want to create a metric panel to track external contributions to the PyTorch repo

This PR creates a daily job to track how many external contributions occurred the day before and uploads it to a s3 collection which is accessible by rockset.

`upload_external_contrib_stats.py` is a python script which grabs the neccesary stats from github and sticks them into an s3 bucket. It is used here to do daily uploads, but can generally be used for larger queries as well.


[ghstack-poisoned]

PaliC committed:
Update on "Upload external contribution data to s3"


Context: We want to create a metric panel to track external contributions to the PyTorch repo

This PR creates a daily job to track how many external contributions occurred the day before and uploads it to a s3 collection which is accessible by rockset.

`upload_external_contrib_stats.py` is a python script which grabs the neccesary stats from github and sticks them into an s3 bucket. It is used here to do daily uploads, but can generally be used for larger queries as well.


[ghstack-poisoned]

PaliC committed:
Update on "Upload external contribution data to s3"


Context: We want to create a metric panel to track external contributions to the PyTorch repo

This PR creates a daily job to track how many external contributions occurred the day before and uploads it to a s3 collection which is accessible by rockset.

`upload_external_contrib_stats.py` is a python script which grabs the neccesary stats from github and sticks them into an s3 bucket. It is used here to do daily uploads, but can generally be used for larger queries as well.


[ghstack-poisoned]

PaliC commented:
@pytorchbot merge -g

PaliC commented:
@pytorchbot merge -f "broken tests are for inductor which is unrelated to these changes"

clee2000 commented:
@pytorchbot revert -m "broke lint on master, merge base is too old, https://github.com/pytorch/pytorch/actions/runs/4315881630/jobs/7531170401 https://hud.pytorch.org/pytorch/pytorch/commit/f418e1f8b63c0c15f52b373a57bfd9d65d02b172#11721314649" -c landrace

clee2000 commented:
@pytorchbot rebase

XuehaiPan commented on tools/stats/upload_external_contrib_stats.py:
```diff
@@ -0,0 +1,142 @@
+# import time
+from tools.stats.upload_stats_lib import upload_to_s3
+
+FILTER_OUT_USERS = set(["pytorchmergebot", "facebook-github-bot", "pytorch-bot[bot]"])
```
`flake8-comprehensions` complains about unnecessary literal.

```suggestion
FILTER_OUT_USERS = {"pytorchmergebot", "facebook-github-bot", "pytorch-bot[bot]"}
```

PaliC committed:
Update on "Upload external contribution data to s3"


Context: We want to create a metric panel to track external contributions to the PyTorch repo

This PR creates a daily job to track how many external contributions occurred the day before and uploads it to a s3 collection which is accessible by rockset.

`upload_external_contrib_stats.py` is a python script which grabs the neccesary stats from github and sticks them into an s3 bucket. It is used here to do daily uploads, but can generally be used for larger queries as well.


[ghstack-poisoned]

PaliC committed:
Update on "Upload external contribution data to s3"


Context: We want to create a metric panel to track external contributions to the PyTorch repo

This PR creates a daily job to track how many external contributions occurred the day before and uploads it to a s3 collection which is accessible by rockset.

`upload_external_contrib_stats.py` is a python script which grabs the neccesary stats from github and sticks them into an s3 bucket. It is used here to do daily uploads, but can generally be used for larger queries as well.


[ghstack-poisoned]
